lib/clearance/user.rb

```diff
@@ -103,12 +100,34 @@ module User
     extend ActiveSupport::Concern
 
     included do
-      attr_accessor :password_changing
       attr_reader :password
 
       include Validations
       include Callbacks
       include password_strategy
+
+      def password=(value)
+        encrypted_password_will_change!
+        super
+      end
+
+      def password_changing
+        warn "#{Kernel.caller.first}: [DEPRECATION] " \
+          "The `password_changing` attribute is deprecated. Clearance uses " \
+          "the dirty state of the `encrypted_password` field to track this " \
+          "automatically."
+
+        @password_changing
```

Comment:
I'd have to keep the `attr_accessor` in that case, and I find declaring the accessor that I'm going to immediately override to be more confusing than inlining the equivalent functionality.